### PR TITLE
[ISSUE #3160]fix compatibility issue when reading cache files with suffix ".datum"

### PR DESCRIPTION
**Pull request by Maijh97**
Please do not create a Pull Request without creating an issue first.

## What is the purpose of the change

fix: #3160 

## Brief changelog

XX

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily:

* [ ] Make sure there is a Github issue filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
* [ ] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
* [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
* [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/alibaba/nacos/tree/master/test).
* [ ] Run `mvn -B clean package apache-rat:check findbugs:findbugs -Dmaven.test.skip=true` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.

### CODE REVIEW

File: naming/src/main/java/com/alibaba/nacos/naming/consistency/persistent/raft/RaftStore.java [hunk 1]
```diff
@@ -155,13 +157,15 @@ private synchronized Datum readDatum(File file, String namespaceId) throws IOExc
             if (StringUtils.isBlank(json)) {
                 return null;
             }
+            JsonNode jsonNode = JacksonUtils.toObj(json);
```
Comment (KomachiSion) [inline naming/src/main/java/com/alibaba/nacos/naming/consistency/persistent/raft/RaftStore.java:160]: I think still should use file name to identify

File: naming/src/main/java/com/alibaba/nacos/naming/consistency/persistent/raft/RaftStore.java [hunk 2]
```diff
@@ -139,7 +141,7 @@ public synchronized Datum load(String key) throws Exception {
     }
     
     private boolean isDatumCacheFile(String fileName) {
-        return fileName.endsWith(CACHE_FILE_SUFFIX);
+        return fileName.endsWith(RAFT_CACHE_FILE_SUFFIX) || fileName.startsWith(RAFT_CACHE_FILE_PREFIX);
```
Comment (KomachiSion) [inline naming/src/main/java/com/alibaba/nacos/naming/consistency/persistent/raft/RaftStore.java:144]: I think the second condition maybe too wide. should be add `.naming` for it?

File: naming/src/main/java/com/alibaba/nacos/naming/consistency/persistent/raft/RaftStore.java (hunk 1 above)
Comment (Maijh97) [reply naming/src/main/java/com/alibaba/nacos/naming/consistency/persistent/raft/RaftStore.java:160]: Okay

File: naming/src/main/java/com/alibaba/nacos/naming/consistency/persistent/raft/RaftStore.java (hunk 2 above)
Comment (Maijh97) [reply naming/src/main/java/com/alibaba/nacos/naming/consistency/persistent/raft/RaftStore.java:144]: Okay

**Comment by Maijh97** [comment]
I have adjusted the code.please help to review this pr again, thanks. @KomachiSion

**Pull request merged**